code/game/objects/structures/crates_lockers/crates.dm

```diff
@@ -331,3 +331,99 @@
 	icon_state = "closed_mcart_y"
 	icon_opened = "open_mcart_y"
 	icon_closed = "closed_mcart_y"
+
+/obj/structure/closet/crate/mass_produced_crate
+	name = "Mass Produced Crate"
+	desc = "A rectangular steel crate. Cannot be welded for metal."
+	icon = 'icons/obj/structures/crates.dmi'
+	icon_state = "closed_basic"
+	icon_opened = "open_basic"
+	icon_closed = "closed_basic"
+	anchored = FALSE
+	mob_storage_capacity = 0
+	max_integrity = 5 //hopefully makes it so req crate spam can easily be destroyed
+	closet_flags = CLOSET_ALLOW_OBJS|CLOSET_ALLOW_DENSE_OBJ
+	density = TRUE
+	materials = list(/datum/material/metal = 200)
+
+/obj/structure/closet/crate/mass_produced_crate/welder_act(mob/living/user, obj/item/tool/weldingtool/welder) //code that hopefully handles if crate drops metal when welded
+	if(!welder.isOn())
+		return FALSE
+
+	if(opened)
+		if(!welder.use_tool(src, user, 2 SECONDS, 1, 50))
+			to_chat(user, span_notice("You need more welding fuel to complete this task."))
+			return TRUE
+		visible_message(span_notice("\The [src] has been cut apart by [user] with [welder]."), "You hear welding.")
+		qdel(src)
+		return TRUE
```

Comment:
This is not the way to do this, its just copy pasta code.
I'd recommend making the dropped material a var at the closet level, then you can just make the var null for this crate.

Reply:
after alot of trial error I realized that I am too stupid to make this happen

can you help me figure out how to turn this into a variable?